RATIONALE.md

```diff
@@ -0,0 +1,49 @@
+# Rationale for GitOps
+
+Currently, many software system's desired state is not defined separately from the running system.
+When we desire the behaviour of a software system to change, we modify the system's configuration directly, either through human action, or by running scripts that take a set of predetermined action on the system.
+
+This leads to several serious issues:
+
+- **Detecting drift from desired state**
+
+    If the desired state of a system is not explicitly defined, it is impossible to verify if the system in a correct state. The state of a running system itself does not provide sufficent information to determine its correctness. 
+    
+    Consider logging into an administration console and seeing that 28 machines are healthily running.
```

Comment:
I think we should look for an example more consistent with cloud native architecture and operational practices, this is not a realistic, dynamic horizontally scaling scenario of ephemeral resources, which would most likely be more applicable to the use cases that will be looking to adopt GitOps.

Reply:
I do think this is still reality. And when "selling" those low-tech firms the cloud-native cool-aid, GitOps-like practices are one of the showcases that really go well.
I agree another example for existing cloud architectures would make sense, too (as an addition).

Reply:
Concur. A better example would be a welcome here.

Reply:
@murillodigital do you want to propose an additional, cloud native example?

Reply:
How about we consider the following example authored by @TomOnTime?

Thank you @murillodigital @tonit @bricef @scottrigby @moshloop @TomOnTime.
```suggestion
    An excerpt from https://queue.acm.org/detail.cfm?id=3237207

        For example, a desired state for a VM configuration system might state, "There should be three virtual machines named foo1, foo2, and foo3." When the file is processed the first time, the three VMs are created. Processing the same configuration file a second time will leave the system unchanged, as the machines already exist. The configuration file is idempotent. If for some reason foo2 were deleted, processing the file again would re-create foo2.

        Contrast this with an imperative configuration language that states, "Add three new virtual machines." This would generate three additional machines every time it is run, eventually exhausting all resources.
```